content/ui-patterns/loading.mdx

```diff
@@ -0,0 +1,406 @@
+</DoDontContainer>
+
+### Navigation
+
+#### Sequential and related groups of navigation items
+
+<img
+  width="960"
+  alt="3 images. 1. Repos and teams loading; 2: Repos and teams both failed; 3: Repos and teams both succeeded"
+  src="https://github.com/primer/design/assets/2313998/ba52e203-a0fa-4a55-a592-f436ffc2fda3"
+/>
+
+If sequential groups or navigation links are unable to load, replace both groups with a single loading indicator and a single error message above the [nav list](components/nav-list).
+
+#### Non-sequential groups of navigation items
+
+<img
+  width="960"
+  alt="3 images. 1. Repos, teams, and organizations loading; 2: Repos and teams both failed; 3: All groups succeeded"
+  src="https://github.com/primer/design/assets/2313998/14cbb97d-ebd2-44ca-b434-56454c9f6d47"
+/>
+
+<Note>
+  This example is for demonstration purposes only. At the time these guidelines were written, there is no
+  "Organizations" group, and teams and repositories data are loaded via a single controller.
+</Note>
+
+If non-sequential groups of navigation links are in a loading state, preserve the group headings and replace the contents of each group with a single loading indicator. If loading fails, show a single error message above the [nav list](components/nav-list).
+
+#### Individual navigation items
+
+<img
+  width="960"
+  alt="3 images. 1. Some items in right side-sheet are loading; 2: Some items in right side-sheet failed; 3: All items in right side-sheet succeeded"
+  src="https://github.com/primer/design/assets/2313998/1b619fa0-0870-4356-a5a6-78519a36f539"
+/>
+
+If some links in the right side sheet cannot be loaded, default to showing each [nav list](components/nav-list) item in an inert state. The users should still be able to hover, click, and focus inert items, but they don't link to anywhere. Optionally, a tooltip may be displayed explaining that the navigation item is unavailable. An error banner is not necessary for this case.
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  justifyContent="space-between"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 3}}
+>
+
+<div>
+
+If we don't know specifically which links failed to load (for example, some users will never see the "Your enterprises" link), then hide the links and show a generic error message above the [nav list](components/nav-list).
+
+</div>
+
+<img
+  width="456"
+  alt="Generic error banner with hidden items"
+  src="https://github.com/primer/design/assets/2313998/a6275e1c-72b8-49b0-a9cf-689ec763a000"
+/>
+
+</Box>
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  justifyContent="space-between"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 3}}
+>
+
+<div>
+
+Default to replacing each loading navigation item with the item in a loading state. If we know what the link's label is, show it. If we don't know what the link's label is, show generic loading text in its place.
```

Comment:
I might be reading this incorrectly, but I think this section contradicts this section above:

> If non-sequential groups of navigation links are in a loading state, preserve the group headings and replace the contents of each group with a single loading indicator.

Since its showing several loading items next to each other rather than a single loader?

Reply:
I'm suggesting that we show a single loader for sequential _groups_, not sequential _items_. I totally see the confusion though, and I'll make it more clear.